src/main/java/echoquery/sql/QueryBuilder.java

```diff
@@ -64,43 +63,46 @@ public Query build() {
   }
 
   public static QueryBuilder of(Intent intent) {
-    String table = intent.getSlot(SlotNames.TABLE_NAME).getValue();
+    String table = intent.getSlot(SlotUtil.TABLE_NAME).getValue();
     SchemaInferrer inferrer = SchemaInferrer.getInstance();
 
-    String column = intent.getSlot(SlotNames.COLUMN_NAME).getValue();
-    String colVal= intent.getSlot(SlotNames.COLUMN_VALUE).getValue();
-    String colNum = intent.getSlot(SlotNames.COLUMN_NUMBER).getValue();
-    String equals = intent.getSlot(SlotNames.EQUALS).getValue();
-    String greater = intent.getSlot(SlotNames.GREATER_THAN).getValue();
-    String less = intent.getSlot(SlotNames.LESS_THAN).getValue();
+    String aggregate = intent.getSlot(SlotUtil.AGGREGATE).getValue();
+    String aggregationColumn =
+        intent.getSlot(SlotUtil.AGGREGATION_COLUMN).getValue();
+    String comparisonColumn =
+        intent.getSlot(SlotUtil.COMPARISON_COLUMN).getValue();
+    String comparator = intent.getSlot(SlotUtil.COMPARATOR).getValue();
+    String colVal= intent.getSlot(SlotUtil.COLUMN_VALUE).getValue();
+    String colNum = intent.getSlot(SlotUtil.COLUMN_NUMBER).getValue();
+
+    QueryBuilder builder = new QueryBuilder();
+
+    Expression aggregationFunc;
+    if (aggregationColumn == null) {
+      aggregationFunc = QueryUtil.functionCall(
+          SlotUtil.getAggregateFunction(aggregate));
+    } else {
+      aggregationFunc = QueryUtil.functionCall(
+          SlotUtil.getAggregateFunction(aggregate),
+          QueryUtil.nameReference(aggregationColumn));
```

Comment:
that isn't always true though right? from is rendered from the comparison column referencing table B - but it could be possible that someone is aggregating over a column from table A? So it wouldnt make sense if they both were B.aggregateColumn, and B.comparisonColumn - unless does wherePrefix() do something more sophisticated?